src/Corruptions.ts

```diff
@@ -204,69 +202,114 @@ export const corruptionButtonsAdd = () => {
     }
 }
 
-export const corruptionLoadoutTableCreate = () => {
-    const corrCount = 8
-    const table = getElementById<HTMLTableElement>('corruptionLoadoutTable')
+const loadoutTemplate = (player: Player) => {
+    const loadouts = Object.values(player.corruptionLoadouts)
+        .map((corruptions, i) => ({
+            corruptions,
+            name: player.corruptionLoadoutNames[i]
+        }))
 
-    for (let i = 0; i < Object.keys(player.corruptionLoadouts).length + 1; i++) {
-        const row = table.insertRow()
-        for (let j = 0; j <= corrCount; j++) {
-            const cell = row.insertCell();
-            cell.className = `test${j}`
-            if (j === 0) {
-                if (i === 0) {
-                    cell.textContent = 'Next:'
-                }
-                // Other loadout names are updated after player load in Synergism.ts > loadSynergy
-            } else if (j <= corrCount) {
-                // start two-indexed (WTF!)
-                cell.textContent = ((i === 0) ? player.prototypeCorruptions[j+1] : player.corruptionLoadouts[i][j+1]).toString();
-                cell.style.textAlign = 'center'
-            }
-        }
-        if (i === 0) {
-            let cell = row.insertCell();
-            //empty
-
-            cell = row.insertCell();
-            const btn = document.createElement('button');
-            btn.className = 'corrLoad'
-            btn.textContent = 'Zero'
-            btn.onclick = () => corruptionLoadoutSaveLoad(false, i);
-            cell.appendChild(btn);
-            cell.title = 'Reset corruptions to zero on your next ascension'
-        } else {
-            let cell = row.insertCell();
-            let btn = document.createElement('button');
-            btn.className = 'corrSave'
-            btn.textContent = 'Save'
-            btn.onclick = () => corruptionLoadoutSaveLoad(true, i);
-            cell.appendChild(btn);
+    return `
+      <table id="corruptionLoadoutTable" alt="corruptionLoadoutTable" label="corruptionLoadoutTable">
```

Comment:
Not so optimal to just replace-it-all-in-place on each "update". But I think this should be fine for this view. Manually keeping track of updating the DOM seems more annoying. (this is where I miss a small lightweight react-like library :P)